uaa/src/test/java/org/cloudfoundry/identity/uaa/mock/providers/IdentityProviderEndpointsAliasMockMvcTests.java

```diff
@@ -327,229 +373,351 @@ protected UpdateBase(final boolean aliasFeatureEnabled) {
-            }
+                @Test
+                void shouldAccept_OtherPropertiesOfIdpWithAliasAreChanged_CustomToUaaZone() throws Exception {
+                    shouldAccept_OtherPropertiesOfIdpWithAliasAreChanged(customZone, IdentityZone.getUaa());
+                }
 
-            private static Stream<Arguments> shouldReject_ChangingAliasPropertiesOfIdpWithAlias() {
-                return Stream.of(null, "", "other").flatMap(aliasIdValue ->
-                        Stream.of(null, "", "other").map(aliasZidValue ->
-                                Arguments.of(aliasIdValue, aliasZidValue)
-                        ));
-            }
+                private void shouldAccept_OtherPropertiesOfIdpWithAliasAreChanged(final IdentityZone zone1, final IdentityZone zone2) throws Exception {
+                    // create an IdP with an alias
+                    final IdentityProvider<?> originalIdp = createIdpWithAlias(zone1, zone2);
+
+                    // update other property
+                    final String newName = "new name";
+                    originalIdp.setName(newName);
+                    final IdentityProvider<?> updatedOriginalIdp = updateIdp(zone1, originalIdp);
+                    assertThat(updatedOriginalIdp).isNotNull();
+                    assertThat(updatedOriginalIdp.getAliasId()).isNotBlank();
+                    assertThat(updatedOriginalIdp.getAliasZid()).isNotBlank();
+                    assertThat(updatedOriginalIdp.getAliasZid()).isEqualTo(zone2.getId());
+                    assertThat(updatedOriginalIdp.getName()).isNotBlank().isEqualTo(newName);
+
+                    // check if the change is propagated to the alias IdP
+                    final String id = updatedOriginalIdp.getAliasId();
+                    final Optional<IdentityProvider<?>> aliasIdp = readIdpFromZoneIfExists(zone2.getId(), id);
+                    assertThat(aliasIdp).isPresent();
+                    assertIdpReferencesOtherIdp(aliasIdp.get(), updatedOriginalIdp);
+                    assertThat(aliasIdp.get().getName()).isNotBlank().isEqualTo(newName);
+
+                    // check if both have the same non-empty relying party secret in the DB
+                    assertIdpAndAliasHaveSameRelyingPartySecretInDb(updatedOriginalIdp);
+
+                    // check if the returned IdP has a redacted relying party secret
+                    assertRelyingPartySecretIsRedacted(updatedOriginalIdp);
+                }
 
-            @Test
-            void shouldReject_AliasNotSupportedForIdpType_UaaToCustomZone() throws Exception {
-                shouldReject_AliasNotSupportedForIdpType(IdentityZone.getUaa(), customZone);
-            }
+                @Test
+                void shouldReject_AliasIdNotSetInPayload_UaaToCustomZone() throws Exception {
+                    shouldReject_AliasIdNotSetInPayload(IdentityZone.getUaa(), customZone);
+                }
```

Comment:
Looks good. Leaving comments to track my progress as this is challenging to read through.